packages/core/src/utils.ts

```diff
@@ -93,12 +93,14 @@ export async function copyWalletAddress(text: string): Promise<void> {
   }
 }
 
+export const decimalToHex = (decimal: number): string => `0x${decimal.toString(16)}`
```

Comment:
Ok so to keep things DRY, we can create a function that will take a number or string and ensure that it is a hex string:

```typescript
export const toHexString = (val: number | string): string => typeof val === 'number' ? `0x${decimal.toString(16)}` : val
```

Then we can use that in the `chainIdToHex` func:

```typescript
export function chainIdToHex(chains : Chain[] | ChainWithDecimalId[] ): 
Chain[] {
  return chains.map(({ id, ...rest }) => { 
    const hexId = toHexString(id)
    return { id: hexId, ...rest }
  })  
} 
```

We can then replace line 24 in src/chain.ts with:

```javascript
const chainIdHex = toHexString(chainId)
```